app/routes.py

```diff
@@ -1,2 +1,225 @@
-from flask import Blueprint
+from flask import Blueprint, request, make_response, jsonify
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import datetime
+from sqlalchemy import DateTime, desc
+import requests
+import os
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+#wave4
+def hi_slack_api(task):
+    key = os.environ.get("API_KEY")
+    url = "https://slack.com/api/chat.postMessage"
+    slack_str = f"Someone just completed the task {task.title}"
+    requests.post(url, headers={"Authorization": key}, params={"channel": "task-list-api", "text": slack_str})
+
+#wave6
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes= False)
+def task_goal(goal_id):
+    request_body = request.get_json()
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("", 404)
+
+    for task_id in request_body["task_ids"]:
+        task = Task.query.get(task_id)
+        task.goal_id = goal.goal_id
+ 
+    db.session.commit()   
```

Comment:
Good job remembering to commit here! Many folks forgot to since there isn't actually a test to verify the new tasks get saved in the db